### ISSUE TITLE
Crash in v0.0.3

### ISSUE THREAD

**Issue by philipheinser** [prebid/prebid-mobile-ios#24]
You are looping over an array in your library while at the same time updating it on another threat thread. We ran into this in production and now have to deal with massive crashes thanks to this lib -.-

```
Crashed: NSOperationQueue 0x174a24820 :: NSOperation 0x170e5ba50 (QOS: DEFAULT)
0  libobjc.A.dylib                0x184bf4150 objc_msgSend + 16
1  CoreFoundation                 0x18605b45c -[__NSCFString isEqual:] + 56
2  CoreFoundation                 0x18607512c -[NSMutableArray removeObject:] + 152
3  promiflash                     0x1003f798c -[PBServerFetcher processData:] (PBServerFetcher.m:88)
4  promiflash                     0x1003f753c __56-[PBServerFetcher makeBidRequest:withCompletionHandler:]_block_invoke (PBServerFetcher.m:56)
5  CFNetwork                      0x186868780 __67+[NSURLConnection sendAsynchronousRequest:queue:completionHandler:]_block_invoke_2 + 180
6  Foundation                     0x186c4e854 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
7  Foundation                     0x186b937b0 -[NSBlockOperation main] + 96
8  Foundation                     0x186b83b68 -[__NSOperationInternal _start:] + 612
9  Foundation                     0x186c50bf0 __NSOQSchedule_f + 228
10 libdispatch.dylib              0x1850369a0 _dispatch_client_callout + 16
11 libdispatch.dylib              0x185044ad4 _dispatch_queue_serial_drain + 928
12 libdispatch.dylib              0x18503a2cc _dispatch_queue_invoke + 884
13 libdispatch.dylib              0x185046a50 _dispatch_root_queue_drain + 540
14 libdispatch.dylib              0x1850467d0 _dispatch_worker_thread3 + 124
15 libsystem_pthread.dylib        0x18523f100 _pthread_wqthread + 1096
16 libsystem_pthread.dylib        0x18523ecac start_wqthread + 4
```

**Comment by nhedley** [comment]
Hi @philipheinser, I'm so sorry you've experienced that crash. We are aware of it and a fix was merged earlier this week (https://github.com/prebid/prebid-mobile-ios/pull/19). Master version should not have the crash and we will release 0.0.4 with the fix this coming week.

**Comment by nhedley** [comment]
We just released version 0.0.4 with the fix for this crash available on Github and Cocoapods.
